sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADResourceServerClientConfiguration.java

```diff
@@ -37,58 +40,87 @@
 @EnableConfigurationProperties({ AADAuthenticationProperties.class })
 @ConditionalOnClass({ BearerTokenAuthenticationToken.class, OAuth2LoginAuthenticationFilter.class })
 @ConditionalOnProperty(prefix = "azure.activedirectory", value = "client-id")
-public class AADResourceServerOboConfiguration {
+public class AADResourceServerClientConfiguration {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AADResourceServerOboConfiguration.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(AADResourceServerClientConfiguration.class);
 
     @Autowired
     private AADAuthenticationProperties properties;
 
     @Bean
     @ConditionalOnMissingBean({ ClientRegistrationRepository.class })
     public ClientRegistrationRepository clientRegistrationRepository() {
-        final List<ClientRegistration> oboClients = createOboClients();
+        final List<ClientRegistration> oboClients = createClients();
         if (oboClients.isEmpty()) {
-            LOGGER.warn("No client registrations are found for AAD Obo.");
+            LOGGER.warn("No client registrations are found for AAD Client.");
             return registrationId -> null;
         }
         return new InMemoryClientRegistrationRepository(oboClients);
     }
 
+    @Bean
+    @ConditionalOnMissingBean
+    OAuth2AuthorizedClientService authorizedClientService(ClientRegistrationRepository clientRegistrationRepository) {
+        return new InMemoryOAuth2AuthorizedClientService(clientRegistrationRepository);
+    }
+
     /**
-     * Use InMemoryClientRegistrationRepository to create AADOAuth2OboAuthorizedClientRepository
+     * Use InMemoryClientRegistrationRepository to create AADResourceServerOAuth2AuthorizedClientRepository
      *
      * @param repo client registration
-     * @return AADOAuth2OboAuthorizedClientRepository Bean
+     * @return AADResourceServerOAuth2AuthorizedClientRepository Bean
      */
     @Bean
     @ConditionalOnMissingBean
-    public OAuth2AuthorizedClientRepository oAuth2AuthorizedClientRepository(ClientRegistrationRepository repo) {
-        return new AADOAuth2OboAuthorizedClientRepository(repo);
+    public OAuth2AuthorizedClientRepository oAuth2AuthorizedClientRepository(
+        ClientRegistrationRepository repo,
+        OAuth2AuthorizedClientService oAuth2AuthorizedClientService) {
+        return new AADResourceServerOAuth2AuthorizedClientRepository(oAuth2AuthorizedClientService, repo);
     }
```

Comment:
What's the difference between `OAuth2AuthorizedClientRepository` and `OAuth2AuthorizedClientService`? Why are we injecting two beans here?

Reply:
`OAuth2AuthorizedClientService` : Responsible for the management of Authorized Client(s), which provide the purpose of associating an Access Token credential to a Client and Resource Owner.

`ClientRegistrationRepository` : A repository for ClientRegistration(s).It simply provides the `findByRegistrationID` method.

Reply:
In `OAuth2AuthorizedClientRepository` didn't go to the operation of the storage `authorizedclient`, the real storage `authorizedclient` operation is to `OAuth2AuthorizedClientService` to do.